engine/resource/src/resource.cpp

```diff
@@ -119,6 +119,9 @@ struct SResourceFactory
     dmResourceMounts::HContext                   m_Mounts;
     dmResourceProvider::HArchive                 m_BuiltinMount;
     dmResourceProvider::HArchive                 m_BaseArchiveMount;
+
+    // Serial version that increases per resource insertion
```

Comment:
Hmm, I think it should count up for each recreate as well. E.g. hot reloading a new asset.